mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt

```diff
@@ -213,26 +218,65 @@ class WidgetListFragment : Fragment(), WidgetAdapter.ItemClickListener {
-                    populateNfcStatesMenu(nfcMenu, context, suggestedCommands, widget)
+                    populateStatesMenu(nfcMenu, context, suggestedCommands) { state, mappedState ->
+                        startActivity(WriteTagActivity.createItemUpdateIntent(
+                            context,
+                            widget.item?.name ?: return@populateStatesMenu,
+                            state,
+                            mappedState,
+                            widget.label)
+                        )
+                    }
                 }
                 menu.add(Menu.NONE, CONTEXT_MENU_ID_WRITE_SITEMAP_TAG, Menu.NONE, R.string.nfc_action_to_sitemap_page)
             }
+            if (hasCommandOptions) {
+                val widgetMenu = menu.addSubMenu(Menu.NONE, CONTEXT_MENU_ID_CREATE_HOME_SCREEN_WIDGET, Menu.NONE,
+                    R.string.create_home_screen_widget_title)
+                widgetMenu.setHeaderTitle(R.string.item_picker_dialog_title)
+                populateStatesMenu(widgetMenu, context, suggestedCommands) { state, mappedState ->
+                    requestPinAppWidget(context, widget, state, mappedState)
+                }
+            }
             if (ShortcutManagerCompat.isRequestPinShortcutSupported(context)) {
                 menu.add(Menu.NONE, CONTEXT_MENU_ID_PIN_HOME, Menu.NONE, R.string.home_shortcut_pin_to_home)
             }
-        } else if (hasCommandOptions && nfcSupported) {
-            menu.setHeaderTitle(R.string.nfc_action_write_command_tag)
-            populateNfcStatesMenu(menu, context, suggestedCommands, widget)
+        } else if (hasCommandOptions) {
+            if (nfcSupported) {
+                val nfcMenu = menu.addSubMenu(Menu.NONE, CONTEXT_MENU_ID_WRITE_ITEM_TAG, Menu.NONE,
+                    R.string.nfc_action_write_command_tag)
+                nfcMenu.setHeaderTitle(R.string.item_picker_dialog_title)
+                populateStatesMenu(nfcMenu, context, suggestedCommands) { state, mappedState ->
+                    startActivity(WriteTagActivity.createItemUpdateIntent(
+                        context,
+                        widget.item?.name ?: return@populateStatesMenu,
+                        state,
+                        mappedState,
+                        widget.label)
+                    )
+                }
+
+                val widgetMenu = menu.addSubMenu(Menu.NONE, CONTEXT_MENU_ID_CREATE_HOME_SCREEN_WIDGET, Menu.NONE,
+                    R.string.create_home_screen_widget_title)
+                widgetMenu.setHeaderTitle(R.string.item_picker_dialog_title)
+                populateStatesMenu(widgetMenu, context, suggestedCommands) { state, mappedState ->
+                    requestPinAppWidget(context, widget, state, mappedState)
+                }
+            } else {
+                menu.setHeaderTitle(R.string.create_home_screen_widget_title)
```

Comment:
Dropping this case and instead having a context menu with one submenu would allow a simplification of the code.